dashboard/app/views/levels/editors/fields/_default_song.html.haml

```diff
@@ -6,7 +6,7 @@
   It is defined as: If there are selected songs, then use the first one in the list. Otherwise, use the first song in the list
   of all songs.
 - songMap = @level.class.hoc_songs.reduce({}) {|hash, song| hash[song[1]] = song; hash }
-- selectedSongObjects = @level.song_selection ? @level.song_selection.map {|s| songMap[s] } : @level.class.hoc_songs
+- selectedSongObjects = @level.song_selection ? @level.song_selection.map {|s| songMap[s] }.select{|s| s} : @level.class.hoc_songs
```

Comment:
Also not an expert, but I think `.compact` might be the more standard to remove nil values in Ruby? [Docs](https://ruby-doc.org/core-1.9.3/Array.html#method-i-compact)

Reply:
That's the kinda ruby knowledge I lack! tyvm - changed it.